### PR TITLE
Add support for factoryFor

### PR DESCRIPTION
**Pull request by chadhietala**
Prep work for https://github.com/emberjs/ember.js/pull/14360

### CODE REVIEW

**Comment by Turbo87** [comment]
@chadhietala I suppose we can't test this properly until the other PR is merged, correct?

File: lib/ember-test-helpers/build-registry.js [hunk 1]
```diff
@@ -30,12 +30,31 @@ function exposeRegistryMethodsWithoutDeprecations(container) {
...
+    return this.__container__[LOOKUP_FACTORY](...arguments);
+  }
+});
+
+function buildOwner() {
+  if (Ember._RegistryProxyMixin && Ember._ContainerProxyMixin && FACTORY_FOR && LOOKUP_FACTORY) {
+    return Ember.Object.extend(Ember._RegistryProxyMixin, Ember._ContainerProxyMixin, FactoryForMixin);
+  } else if (Ember._RegistryProxyMixin && Ember._ContainerProxyMixin) {
     return Ember.Object.extend(Ember._RegistryProxyMixin, Ember._ContainerProxyMixin);
+  } else if (FACTORY_FOR && LOOKUP_FACTORY) {
```
Comment (rwjblue) [inline lib/ember-test-helpers/build-registry.js]: Can this case actually happen?

**Comment by rwjblue** [comment]
Given my suggestion in https://github.com/emberjs/ember.js/pull/14360#pullrequestreview-13465262, no changes would be needed here in ember-test-helpers.

**Comment by chadhietala** [comment]
Just going to put this stuff on the `ContainerProxy` mixin

**Pull request closed without merging**